### ISSUE TITLE
[question] What is the battery duration on the 4K 9570 version?

### ISSUE THREAD

**Issue by toonvanstrijp** [LuletterSoul/Dell-XPS-15-9570-macOS-Mojave#32]
Last time I tried this repository I had a lot of troubles to get the battery to perform well. That time my battery lasted for 2 hours... and the fans where also spinning constantly. Is this repository well suited for a daily usage?

**Comment by LuletterSoul** [comment]
Not so well suited for a daily usage, I got about 2.5 hours battery duration on 4k model.
What the longest battery duration can you get from other repositories?

**Comment by toonvanstrijp** [comment]
I read somewhere that the 1080p model got a battery duration for 7 hours.

I was just wondering why we lose so much battery cause in linux I got 8 hours or more with the 4k model. Maybe the graphics card isn't disabled properly?

**Comment by LuletterSoul** [comment]
I have no idea.The most possible thing that can cause high energy consumption should be the discrete card.I will do more tests and figure out how `SSDT-Disable-DGPU.xml` works.

**Comment by LuletterSoul** [comment]
Finally, I found `SSDT-Disable-DGPU.xml` it do work to disable the discrete graphics card, so disappointing, it seems to be not related to battery usage.

**Comment by chinmayjade** [comment]
I get around 2 to 2.5 hours on the 4K model and have fans constantly running. 
Undervolting by 105mV (i7) using Volta seems to have helped a bit. Activity monitor shows around 5 hours estimate at 100% charge. Don't know if it would last that long though. I will test for a couple days.

**Comment by toonvanstrijp** [comment]
Awesome. I hope we can tweak it a bit so that it runs much closer as Ubuntu does. Cause there I have 11 hours on a full battery :).

**Comment by LuletterSoul** [comment]
The new version could gain 5+ hours  if the machine executes tasks slightly.But it much far from amazing 10+ battery duration. : ). Anyway, I feel fans become more quiet using current version based on `BIOS 1.11.2`.I haven’t use undervolting mehod yet.

<img width="905" alt="Screen Shot 2019-07-16 at 2 19 02 PM" src="https://user-images.githubusercontent.com/21377419/61592246-5e63f800-ac03-11e9-9c2e-9cd6363a800f.png">
<img width="512" alt="Screen Shot 2019-07-15 at 12 18 07 AM" src="https://user-images.githubusercontent.com/21377419/61592247-62901580-ac03-11e9-934d-0229e1914408.png">

**Comment by LukaJankovic** [comment]
I got much better battery life by switching layout id to 72 instead of 30. See [here](https://github.com/kevinshroff/Dell-XPS-15-9570-macOS-Mojave/issues/3)

**Comment by eamigo86** [comment]
Any progress with this issue?

**Comment by khoa-nv** [comment]
> I got much better battery life by switching layout id to 72 instead of 30. See [here](https://github.com/kevinshroff/Dell-XPS-15-9570-macOS-Mojave/issues/3)

Can you use headphone with layout-id 72?

**Comment by attilagyorffy** [comment]
> I get around 2 to 2.5 hours on the 4K model and have fans constantly running.
> Undervolting by 105mV (i7) using Volta seems to have helped a bit. Activity monitor shows around 5 hours estimate at 100% charge. Don't know if it would last that long though. I will test for a couple days.

Did you actually use Volta? Apparently (according to its site) the undervolting functionality only works with Hashwell and Broadwell (2013-12015) processors so it won't apply to the Caby Lake CPU in the Dell XP 9570.

**Comment by khoa-nv** [comment]
> > I get around 2 to 2.5 hours on the 4K model and have fans constantly running.
> > Undervolting by 105mV (i7) using Volta seems to have helped a bit. Activity monitor shows around 5 hours estimate at 100% charge. Don't know if it would last that long though. I will test for a couple days.
> 
> Did you actually use Volta? Apparently (according to its site) the undervolting functionality only works with Hashwell and Broadwell (2013-12015) processors so it won't apply to the Caby Lake CPU in the Dell XP 9570.

You can use voltageshift to disable turbo boost and limit power

**Comment by chinmayjade** [comment]
> > I get around 2 to 2.5 hours on the 4K model and have fans constantly running.
> > Undervolting by 105mV (i7) using Volta seems to have helped a bit. Activity monitor shows around 5 hours estimate at 100% charge. Don't know if it would last that long though. I will test for a couple days.
> 
> Did you actually use Volta? Apparently (according to its site) the undervolting functionality only works with Hashwell and Broadwell (2013-12015) processors so it won't apply to the Caby Lake CPU in the Dell XP 9570.

Yes, I used Volta and seems to be working. You have to disable csrutil from recovery as instructed on the faq page.
